lib/SILGen/SILGenStmt.cpp

```diff
@@ -212,7 +212,22 @@ void StmtEmitter::visitBraceStmt(BraceStmt *S) {
       if (auto *S = ESD.dyn_cast<Stmt*>()) {
         if (S->isImplicit()) continue;
       } else if (auto *E = ESD.dyn_cast<Expr*>()) {
-        if (E->isImplicit()) continue;
+        // Optional chaining expressions are wrapped in a structure like.
+        //
+        // (optional_evaluation_expr implicit type='T?'
+        //   (call_expr type='T?'
+        //     (exprs...
+        //
+        // Walk through it to find out if the statement is actually implicit.
+        if (auto *OEE = dyn_cast<OptionalEvaluationExpr>(E)) {
```

Comment:
Are there other implicit expressions that we need to handle here, like coercions?

Reply:
I can't find anything else like this case, but it might be worth it to just catch all of this at once with a walker that digs for non-implicit ASTNodes.  Not sure if it would be worth the cost tho.